doc/source/main-documentation/qibolab.rst

```diff
@@ -0,0 +1,611 @@
+
+For example, let's first define a platform (that we consider to be a single qubit platform) using the ``create`` method presented in :doc:`/tutorials/lab`:
+
+.. code-block::  python
+
+    from qibolab import Platform
+    platform = Platform("my_platform")
+
+Now we connect and start the instruments (note that we, the user, do not need to know which instruments are connected).
+
+.. code-block::  python
+
+    platform.connect()
+    platform.setup()
+    platform.start()
+
+We can easily print some of the parameters of the channels (similarly we can set those, if needed):
+
+.. note::
+   If the get_method does not apply to the platform (for example there is no local oscillator, to TWPA or no flux tunability...) a ``NotImplementedError`` will be raised.
+
+.. code-block::  python
+
+    print(f"Drive LO frequency: {platform.get_lo_drive_frequency(0)}")
+    print(f"Readout LO frequency: {platform.get_lo_readout_frequency(0)}")
+    print(f"TWPA LO frequency: {platform.get_lo_twpa_frequency(0)}")
+    print(f"TWPA LO power: {platform.get_lo_twpa_power(0)}")
+
+    print(f"Qubit bias: {platform.get_bias(0)}")
+    print(f"Qubit attenuation: {platform.get_attenuation(0)}")
+
+Now we can create a simple sequence (again, without explicitly give any qubit specific parameter, defined in the runcard):
+
+.. code-block::  python
+
+   from qibolab.pulses import PulseSequence
+
+   ps = PulseSequence()
+   ps.add(platform.create_RX_pulse(qubit=0, start=0))   # start time is in ns
+   ps.add(platform.create_RX_pulse(qubit=0, start=100))
+   ps.add(platform.create_MZ_pulse(qubit=0, start=200))
+
+Now we can execute the sequence on hardware:
+
+.. code-block::  python
+
+   results = platform.execute_pulse_sequence(ps)
+
+Finally, we can stop instruments and close connections.
+
+.. code-block::  python
+
+    platform.stop()
+    platform.disconnect()
+
+
+Qubits
+------
+
+The :class:`qibolab.qubits.Qubit` class serves as a comprehensive representation of a physical qubit within the Qibolab framework.
+It encapsulates three fundamental elements crucial to qubit control and operation:
+
+- Channels: Physical Connections
+- Parameters: Configurable Properties
+- Native Gates: Quantum Operations
+
+Channels play a pivotal role in connecting the quantum system to the control infrastructure.
+They are optional and encompass distinct types, each serving a specific purpose:
+
+- readout (from controller device to the qubits)
+- feedback (from qubits to controller)
+- twpa (pump to the twpa)
+- drive
+- flux
+- flux_coupler
+
+The Qubit class allows you to set and manage several key parameters that influence qubit behavior.
+These parameters are typically extracted from the runcard during platform initialization.
+Notable settable parameters include:
```

Comment:
I am not sure if we really need to list all these here, since they should be (if we have the proper docstrings) in the API reference.